barretenberg/cpp/src/barretenberg/vm2/simulation/execution.cpp

```diff
@@ -193,6 +193,15 @@ void Execution::rd_copy(ContextInterface& context,
     }
 }
 
+void Execution::rd_size(ContextInterface& context, MemoryAddress dst_addr)
+{
+    constexpr auto opcode = ExecutionOpCode::RETURNDATASIZE;
+    auto& memory = context.get_memory();
+    MemoryValue rd_size = MemoryValue::from<uint32_t>(context.get_last_rd_size());
```

Comment:
Could you confirm we are tag checking RD_SIZE on return/revert?

Reply:
Good catch - revert was missing this!